File: NumberGuessingGame.pas
program numberGuessingGame(input, output);
var
    num, errorguess, guess, guessCount : integer;
    guessString : string;
begin
    randomize;
    writeln('Guess the random num between 1 - 100!');
    num := random(100) + 1;
    guessCount := 0;

    repeat
        repeat
            writeln('Make a guess: ');
            readln(guessString);
            val(guessString, guess, errorguess);
        until errorguess = 0;

        guessCount := guessCount + 1;
        if guess < num then
            writeln('Your guess was too low, try again!')
        else if guess > num then
            writeln('Your guess was too high, try again!');
    until guess = num;

    writeln('You won!');
    writeln('You needed ', guessCount, ' guesses.');
    writeln('Press ENTER to exit . . .');
    readln;
end.

File: NumberGuessingGame.scala
import scala.util.Random


object NumberGuessingGame {

    def main(args: Array[String]) {
        var guess = -1
        var guessCount = 0
        var num =  (new Random()).nextInt(100) + 1

        while (guess != num) {
            println("Guess the number between 1 - 100: ")
            guess = scala.io.StdIn.readInt()
            guessCount += 1

            if (guess > num) {
                println("Your guess was too high, try again!")
            } else if (guess < num) {
                println("Your guess was too low, try again!")
            } else if (guess == num) {
                println("You win!")
                println("You needed " + guessCount + " guesses to win!")

                println("Press ENTER to exit . . .")
                scala.io.StdIn.readLine()
            }
        }
    }
}
